app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,97 @@
+class TasksController < ApplicationController
+  # TASKS = [
+    # {name: "Call Ma", description: "Family First", completed_at: "", completion_date: ""}
+  # ]
+
+  def index    
+    @tasks = Task.all.order(:created_at)
+  end
+
+  def show
+    task_id = params[:id]
+    @task = Task.find_by(id: task_id)
+    if @task.nil?
+      redirect_to task_path
+    end
+  end
+
+  def new
+    @task = Task.new(name: "Default Name")
+  end
+
+  def create
+    task = Task.new(task_params)
+
+    is_successful = task.save
+
+    if is_successful
+      redirect_to task_path(task.id)
+    else
+      head :not_found #should be a server error instead 
+    end
+  end
+
+  def edit
+    @task = Task.find_by(id: params[:id])
+
+    if @task.nil?
+      redirect_to tasks_path
+    end
+  end
+ 
+  def update
+    task = Task.find_by(id: params[:id])
+
+    if task.nil?
+      redirect_to tasks_path
+    else
+      is_successful = task.update(task_params)
+    end
+
+    if is_successful
+      redirect_to task_path(task.id)
+    end
+  end
+
+  def toggle_complete
+    task = Task.find_by(id: params[:id])
+
+    if task.completed_at == "Incomplete"
```

Comment:
Instead of assuming that `.completed_at` will be a string of either `"Incomplete"` or `"Complete"`, is there a better data type for this?

I think a boolean would be much better, and would prevent bugs. This way you don't need to do `if ... elsif ... else`, but just `if ... else`